---
title: "Final Report: Your Title Here"
author: "Don Francisco"
date: "December 5, 2018"
output:
    github_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Add about 2-3 pages here. Across the whole manuscript, you should cite at least 20 peer reviewed articles.

# Methods

## Study design

The orginial aim [@ahn2018] was to identify the heterogeneity on molecular level with in psoriasis pateints. They did so by using RNA-Seq in order to identify the differences in gene expression as well as flow cytometry to quantify T cell make-up. Their study essentially consisted of isolating skin samples from psoriasis patients and healthy control skin to extract the RNA and analyze the cells via flow cytometry. The design consisted of 3 experimental groups: scalp psoriasis, palmoplantar psoriasis, and conventional plaque psoriasis. A control group was included as well using a healthy control group. Two 4 mm samples were taken from the active psoriatic lesions of the experimental group skin and from the healthy control skin. For each skin sample, 1 biopsy was used for RNA-Seq analysis and the other was used for flow cytometry analysis. The individulas from which the psoriatic skin was obtained were at least 18 years old and were confirmed to have psoriasis for a minnimum of 6 months. Individuals with psoriasis were not on systemic treatments and abstained from topical treatments for at least 2 weeks so as to not interfere with the data. The study design allowed for the comparison of all psoriasis types between eachother as well as with healthy controls 

## Sample origin and sequencing

For the experimental groups, a total of 19 patients with psoriasis were recruited from the San Francisco Bay area. Of these patients, 8 had conventional plaque psoriasis, 8 had scalp psoriasis, and 3 had palmoplantar psoriasis. Two 4 mm punch biopsies were taken from active lesions on the trunk and extremities of the conventional plaque psoriasis, on the scalp of scalp psoriasis patients, and on the hand or foot of the palmoplantar psoriasis patients. For the control group, skin was isolated from the trunk or upperlegs of 9 patients. The healthy control skin was obtained from the surgical discards as a results of cosmetic surgery performed on these patients. The biopsies for RNA-Seq were placed in RNALater to preserve the RNA. The biopsies for flow cytometry were placed on gauze dampened with phosphate buffered saline to keep the skin hydrated. All biopsies were then stored at 4 Â°C until processing of the samples. 

The RNA was extracted from the biopsies strated in RNALater using first the Qiagen Allprep DNA/RNA mini kit. Once extracted, quality control and idetification of mRNA with little degrdation was performed on the Agilent 2100 bioanalyzer in order to filter out any suboptimal mRNA that may interfere with analysis. The quality mRNA that was not discarded was then prepared into a library either the Ovation RNA-Seq System V2 protocol (Nugen) or the Ribozero (Illumina) kits. The prepared libraries were then sequenced on the Illumina Hiseq platform. The reads obtained were paired-end and maintianed a 100 bp length. 


## Computational

These are the methods you used to do your bioinformatic analyses analyses. Should probably be between 0.5 and 1 pages. At a very minimum should include citations for DADA2 and phyloseq if you are doing an amplicon study, or other citations as appropriate.

The data prepared by [@ahn2018] were retrieved from the 

# Results

## Subsections are ok in the results section too

```{r your-code-here, include = FALSE}
# Add code chunks as needed for your analyses
# For most analyses, I would recommend splitting the
# intensive computational part into a seperate R script file
# and then just load your libraries and the data object here
# using the `load()` function. Ask me for clarification if this is
# unclear.
```

```{r load-packages-and-data, include = FALSE}
# load packages as needed
library("plyr")
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("magrittr")
library("tibble")
library("DESeq2")
library("RColorBrewer")
library("ggfortify")
library("cluster")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# load data produced from analysis scripts using
# something like load("output/processed_data.Rdata")
load("output/final_compiled_counts/joined_count_data.RData")

# test that it loaded correctly before proceeding
stopifnot(exists("joined_table"))

# load data produced from differential expression analysis script
load("output/deseq2_data/rld_deseq2_data.RData")

# test that it loaded correctly before proceeding
stopifnot(exists("rld"))

```

In addition to a minimum of 5-8 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion. 2-3 pages.


```{r figure-1-Enrichment-Plot, fig.width = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  mutate(psoriasis_or_healthy = recode(disease_state,
                                       "psoriasis" = "psoriasis",
                                       "palmoplantar psoriasis" = "psoriasis",
                                       "scalp psoriasis" = "psoriasis",
                                       "healthy control" = "healthy")) %>%
  mutate(gene_function = recode(genename,
                                "MKI67" = "Proliferation",
                                "CDK1" = "Proliferation",
                                "CDK2" = "Proliferation",
                                "CDK4" = "Proliferation",
                                "CDK6" = "Proliferation",
                                "CDK7" = "Proliferation",
                                "CDK8" = "Proliferation",
                                "CDK9" = "Proliferation",
                                "CDK3" = "Proliferation",
                                "PCNA" = "Proliferation",
                                "TOP2A" = "Proliferation",
                                "MCM2" = "Proliferation",
                                "MCM3" = "Proliferation",
                                "MCM4" = "Proliferation",
                                "MCM5" = "Proliferation",
                                "MCM6" = "Proliferation",
                                "MCM7" = "Proliferation",
                                "MCM8" = "Proliferation",
                                "MCM9" = "Proliferation",
                                "ITGA7" = "Cell Adhesion Molecules",
                                "ITGB1" = "Cell Adhesion Molecules",
                                "ITGA6" = "Cell Adhesion Molecules",
                                "ITGB2" = "Cell Adhesion Molecules",
                                "ITGAV" = "Cell Adhesion Molecules",
                                "ITGB3" = "Cell Adhesion Molecules",
                                "ITGA2" = "Cell Adhesion Molecules",
                                "ITGA4" = "Cell Adhesion Molecules",
                                "ITGA3" = "Cell Adhesion Molecules",
                                "ITGA5" = "Cell Adhesion Molecules",
                                "ITGA2B" = "Cell Adhesion Molecules",
                                "ITGB4" = "Cell Adhesion Molecules",
                                "ITGAM" = "Cell Adhesion Molecules",
                                "ITGAL" = "Cell Adhesion Molecules",
                                "ITGA1" = "Cell Adhesion Molecules",
                                "ITGAE" = "Cell Adhesion Molecules",
                                "ITGA9" = "Cell Adhesion Molecules",
                                "ITGAX" = "Cell Adhesion Molecules",
                                "ITGB7" = "Cell Adhesion Molecules",
                                "ITGA11" = "Cell Adhesion Molecules",
                                "ITGB6" = "Cell Adhesion Molecules",
                                "ITGA8" = "Cell Adhesion Molecules",
                                "ITGB5" = "Cell Adhesion Molecules",
                                "ITGAD" = "Cell Adhesion Molecules",
                                "ITGA10" = "Cell Adhesion Molecules",
                                "VCAM1" = "Cell Adhesion Molecules",
                                "ICAM1" = "Cell Adhesion Molecules",
                                "ICAM2" = "Cell Adhesion Molecules",
                                "ICAM3" = "Cell Adhesion Molecules",
                                "ICAM4" = "Cell Adhesion Molecules",
                                "ICAM5" = "Cell Adhesion Molecules",
                                "SELP" = "Cell Adhesion Molecules",
                                "SELL" = "Cell Adhesion Molecules",
                                "ICAM5" = "Cell Adhesion Molecules",
                                "CXCR1" = "Chemokine Receptors",
                                "CXCR2" = "Chemokine Receptors",
                                "CXCR3" = "Chemokine Receptors",
                                "CXCR4" = "Chemokine Receptors",
                                "CXCR5" = "Chemokine Receptors",
                                "CXCR6" = "Chemokine Receptors",
                                "CXCR7" = "Chemokine Receptors",
                                "CXCR4" = "Chemokine Receptors",
                                "CCR1" = "Chemokine Receptors",
                                "CCR2" = "Chemokine Receptors",
                                "CCR3" = "Chemokine Receptors",
                                "CCR4" = "Chemokine Receptors",
                                "CCR5" = "Chemokine Receptors",
                                "CCR6" = "Chemokine Receptors",
                                "CCR7" = "Chemokine Receptors",
                                "CCR8" = "Chemokine Receptors",
                                "CCR9" = "Chemokine Receptors",
                                "CCR10" = "Chemokine Receptors",
                                "CCR11" = "Chemokine Receptors",
                                "CXCL1" = "Chemokines",
                                "CXCL2" = "Chemokines",
                                "CXCL3" = "Chemokines",
                                "CXCL4" = "Chemokines",
                                "CXCL5" = "Chemokines",
                                "CXCL6" = "Chemokines",
                                "CXCL7" = "Chemokines",
                                "CXCL8" = "Chemokines",
                                "CXCL9" = "Chemokines",
                                "CXCL10" = "Chemokines",
                                "CXCL11" = "Chemokines",
                                "CXCL12" = "Chemokines",
                                "CXCL13" = "Chemokines",
                                "CXCL14" = "Chemokines",
                                "CXCL15" = "Chemokines",
                                "CXCL16" = "Chemokines",
                                "CCL1" = "Chemokines",
                                "CCL2" = "Chemokines",
                                "CCL3" = "Chemokines",
                                "CCL4" = "Chemokines",
                                "CCL5" = "Chemokines",
                                "CCL6" = "Chemokines",
                                "CCL7" = "Chemokines",
                                "CCL8" = "Chemokines",
                                "CCL9" = "Chemokines",
                                "CCL10" = "Chemokines",
                                "CCL11" = "Chemokines",
                                "CCL12" = "Chemokines",
                                "CCL13" = "Chemokines",
                                "CCL14" = "Chemokines",
                                "CCL15" = "Chemokines",
                                "CCL16" = "Chemokines",
                                "CCL17" = "Chemokines",
                                "CCL18" = "Chemokines",
                                "CCL19" = "Chemokines",
                                "CCL20" = "Chemokines",
                                "CCL21" = "Chemokines",
                                "CCL22" = "Chemokines",
                                "CCL23" = "Chemokines",
                                "CCL24" = "Chemokines",
                                "CCL26" = "Chemokines",
                                "CCL27" = "Chemokines",
                                "CCL28" = "Chemokines",
                                "CCL1" = "Chemokines",
                                "TNF" = "Proinflammatory Cytokines",
                                "TNFRSF1A" =
                                  "Proinflammatory Cytokines Receptors",
                                "TNFRSF1B" =
                                  "Proinflammatory Cytokines Receptors",
                                "TRAF1" = "Proinflammatory Cytokines Receptors",
                                "TRAF2" = "Proinflammatory Cytokines Receptors",
                                "TRAF3" = "Proinflammatory Cytokines Receptors",
                                "TRAF4" = "Proinflammatory Cytokines Receptors",
                                "TRAF5" = "Proinflammatory Cytokines Receptors",
                                "TRAF6" = "Proinflammatory Cytokines Receptors",
                                "IL1B" = "Proinflammatory Cytokines",
                                "IL1A" = "Proinflammatory Cytokines",
                                "IL1R1" = "Proinflammatory Cytokines Receptors",
                                "IL2" = "Proinflammatory Cytokines",
                                "IL2RA" = "Proinflammatory Cytokines Receptors",
                                "IL2RB" = "Proinflammatory Cytokines Receptors",
                                "IL2RG" = "Proinflammatory Cytokines Receptors",
                                "IL6" = "Proinflammatory Cytokines",
                                "IL6R" = "Proinflammatory Cytokines Receptors",
                                "IL12A" = "Proinflammatory Cytokines",
                                "IL12B" =
                                  "Proinflammatory Cytokines",
                                "IL12RB1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL12RB2" =
                                  "Proinflammatory Cytokines Receptors",
                                "IFNG" =
                                  "Proinflammatory Cytokines",
                                "IFNA1" =
                                  "Proinflammatory Cytokines",
                                "IFNGR1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IFNB1" =
                                  "Proinflammatory Cytokines",
                                "IFNAR1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL18" =
                                  "Proinflammatory Cytokines",
                                "IL18R1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL18RAP" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL18BP" = "Proinflammatory Cytokines",
                                "IL7" = "Proinflammatory Cytokines",
                                "IL9" = "Proinflammatory Cytokines",
                                "IL10" = "Proinflammatory Cytokines",
                                "IL7R" = "Proinflammatory Cytokines",
                                "IL9R" = "Proinflammatory Cytokines",
                                "IL10RB" = "Proinflammatory Cytokines",
                                "IL10RA" = "Proinflammatory Cytokines",
                                "IL17A" =
                                  "Proinflammatory Cytokines",
                                "IL17B" =
                                  "Proinflammatory Cytokines",
                                "IL17C" = "Proinflammatory Cytokines",
                                "IL17D" = "Proinflammatory Cytokines",
                                "IL17F" =
                                  "Proinflammatory Cytokines",
                                "IL17RA" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL17RB" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL17RC" =
                                  "Proinflammatory Cytokines Receptors",
                                "CD80" = "Inhibitory Cytokines and Ligands",
                                "CD86" = "Inhibitory Cytokines and Ligands",
                                "CD274" = "Inhibitory Cytokines and Ligands",
                                "BTLA" = "Inhibitory Cytokines and Ligands",
                                "CD160" = "Inhibitory Cytokines and Ligands",
                                "PVAR" = "Inhibitory Cytokines and Ligands",
                                "CD48" = "Inhibitory Cytokines and Ligands",
                                "LGALS9" = "Inhibitory Cytokines and Ligands",
                                "CTLA4" = "Inhibitory Receptors",
                                "PDCD1" = "Inhibitory Receptors",
                                "HAVCR2" = "Inhibitory Receptors",
                                "LAG3" = "Inhibitory Receptors",
                                "TIGIT" = "Inhibitory Receptors",
                                "CD244" = "Inhibitory Receptors")) %>%
  filter(gene_function %in% c("Proliferation",
                              "Cell Adhesion Molecules",
                              "Chemokine Receptors",
                              "Proinflammatory Cytokines Receptors",
                              "Chemokines",
                              "Proinflammatory Cytokines",
                              "Inhibitory Cytokines and Ligands",
                              "Inhibitory Receptors")) %>%
  group_by(gene_function, psoriasis_or_healthy) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(psoriasis_or_healthy, avg_count) %>%
  transform(fold_expression = psoriasis / healthy) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = gene_function,
             y = inverse_or_return,
             fill = gene_function)) +
  geom_col() +
  coord_flip() +
  scale_fill_brewer(palette = "Spectral", name = "Gene Set") +
  geom_hline(yintercept = 0,
                color = "grey35", size = 1) +
  geom_hline(yintercept = -1,
                color = "grey35", size = .5, linetype = "dashed") +
  geom_hline(yintercept = 1,
                color = "grey35", size = .5, linetype = "dashed") +
  ggtitle("Fold Change in Expression of Gene Sets \nbetween Psoriasis and Healthy skin") +
  xlab("Gene Set") +
  ylab("Fold Change")
```

**Figure 1**: 


```{r figure-2-PCA-Plot, fig.align = "center", echo = FALSE}
# Create PCA plot comparing disease type groups based on the top 1000 expressed genes
PCA <- plotPCA(rld, intgroup = c("disease_state"), ntop = 1000, returnData=TRUE)

percentVar <- round(100 * attr(PCA, "percentVar"))

find_hull <- function(PCA) PCA[chull(PCA$PC1, PCA$PC2), ]

hulls <- ddply(PCA, "disease_state", find_hull)

ggplot(PCA, aes(PC1, PC2, color = disease_state)) +
        geom_point(size=3) +
        xlab(paste0("PC1: ",percentVar[1],"% variance")) +
        ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  ggtitle("Comparison of the Psoriasis Disease \nTypes and Healthy Control") +
  geom_polygon(aes(fill = disease_state), data = hulls, alpha = 0.3) +
  guides(fill=FALSE) +
  scale_color_discrete(name = "Condition",
                         labels=c("healthy control" = "Healthy Control",
                            "palmoplantar psoriasis" = "Palmoplantar Psoriasis",
                            "psoriasis" = "Conventional Psoriasis",
                            "scalp psoriasis" = "Scalp Psoriasis"))
```

**Figure 2**: 

```{r figure-3-heatmap-inhibitory-cytokines-and-ligands, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

hm.palette <- colorRampPalette(rev(brewer.pal(9, 'Spectral')), space='Lab')

joined_table %>%
  filter(genename %in% c("CD80",
                         "CD86",
                         "CD274",
                         "BTLA",
                         "CD160",
                         "PVAR",
                         "CD48",
                         "LGALS9")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp = `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type, fold_expression, fold_expression_pp:fold_expression_p) %>%
  select(genename, palmoplantar.psoriasis, psoriasis, scalp.psoriasis, fold_change_type, fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Inhibitory Cytokine \nand Ligand Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change", colours = hm.palette(100), limits = c(-9,10)) +
  scale_x_discrete(labels=c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```


```{r figure-4-heatmap-inhibitory-receptors, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("CTLA4",
                         "PDCD1",
                         "HAVCR2",
                         "LAG3",
                         "TIGIT",
                         "CD244")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp = `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type, fold_expression, fold_expression_pp:fold_expression_p) %>%
  select(genename, palmoplantar.psoriasis, psoriasis, scalp.psoriasis, fold_change_type, fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Inhibitory \nReceptor Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change", colours = hm.palette(100), limits = c(-5,5)) +
  scale_x_discrete(labels=c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```


```{r figure-5-heatmap-chemokine-receptors, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("CXCR1",
                         "CXCR2",
                         "CXCR3",
                         "CXCR4",
                         "CXCR5",
                         "CXCR6",
                         "CXCR7",
                         "CXCR4",
                         "CCR1",
                         "CCR2",
                         "CCR3",
                         "CCR4",
                         "CCR5",
                         "CCR6",
                         "CCR7",
                         "CCR8",
                         "CCR9",
                         "CCR10",
                         "CCR11")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp = `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type, fold_expression, fold_expression_pp:fold_expression_p) %>%
  select(genename, palmoplantar.psoriasis, psoriasis, scalp.psoriasis, fold_change_type, fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Chemokine \nReceptor Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change", colours = hm.palette(100), limit = c(-6.5, 6.5)) +
  scale_x_discrete(labels=c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```


```{r figure-6-heatmap-chemokines, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("CXCL1",
                                "CXCL2",
                                "CXCL3",
                                "CXCL4",
                                "CXCL5",
                                "CXCL6",
                                "CXCL7",
                                "CXCL8",
                                "CXCL9",
                                "CXCL10",
                                "CXCL11",
                                "CXCL12",
                                "CXCL13",
                                "CXCL14",
                                "CXCL15",
                                "CXCL16",
                                "CCL1",
                                "CCL2",
                                "CCL3",
                                "CCL4",
                                "CCL5",
                                "CCL6",
                                "CCL7",
                                "CCL8",
                                "CCL9",
                                "CCL10",
                                "CCL11",
                                "CCL12",
                                "CCL13",
                                "CCL14",
                                "CCL15",
                                "CCL16",
                                "CCL17",
                                "CCL18",
                                "CCL19",
                                "CCL20",
                                "CCL21",
                                "CCL22",
                                "CCL23",
                                "CCL24",
                                "CCL26",
                                "CCL27",
                                "CCL28",
                                "CCL1")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp = `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type, fold_expression, fold_expression_pp:fold_expression_p) %>%
  select(genename, palmoplantar.psoriasis, psoriasis, scalp.psoriasis, fold_change_type, fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Chemokine \nGenes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change", colours = hm.palette(100), limits = c(-40,60)) +
  scale_x_discrete(labels=c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```


```{r figure-7-heatmap-adhesion, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("ITGA7",
                                "ITGB1",
                                "ITGA6",
                                "ITGB2",
                                "ITGAV",
                                "ITGB3",
                                "ITGA2",
                                "ITGA4",
                                "ITGA3",
                                "ITGA5",
                                "ITGA2B",
                                "ITGB4",
                                "ITGAM",
                                "ITGAL",
                                "ITGA1",
                                "ITGAE",
                                "ITGA9",
                                "ITGAX",
                                "ITGB7",
                                "ITGA11",
                                "ITGB6",
                                "ITGA8",
                                "ITGB5",
                                "ITGAD",
                                "ITGA10",
                                "VCAM1",
                                "ICAM1",
                                "ICAM2",
                                "ICAM3",
                                "ICAM4",
                                "ICAM5",
                                "SELP",
                                "SELL",
                                "ICAM5")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp = `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type, fold_expression, fold_expression_pp:fold_expression_p) %>%
  select(genename, palmoplantar.psoriasis, psoriasis, scalp.psoriasis, fold_change_type, fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Cell \nAdhesion Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change", colours = hm.palette(100), limits = c(-6,6)) +
  scale_x_discrete(labels=c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```


# Discussion

Add around 3-4 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited
