---
title: "Cellular Processes Implicated in Psoriasis and their Differences in Psoriasis Subtypes"
author: "Evan Henley"
date: "December 6, 2018"
output:
    github_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Psoriasis is an autoimmune-like disease isolated mainly to the skin that affects around 125 million individuals across the globe [@Griffiths]. The disease manifests itself as itchy and inflamed lesions on the skin that grow excessively. These lesions appear in various parts of the body on individuals and can be so severe that they can cover large portions of the body such as the entire back or on extremities. These lesion can prove quite bothersome to individuals as they may constantly scratch the lesions, further agitating the area and worsening the disease. Moreover, this scratching can even lead to breaking down of the skin barrier, which acts as a first defense against pathogens. By reducing this defense, it allows for a greater chance of pathogen introduction, putting the individual at greater risk for infection [@bartenjev2000]. Older therapies such as Goeckerman therapy and bleach baths, usually used in atopic dermatitis, have been used to treat the symptoms, stopping growth of the skin or killing bacteria that may irritate the lesions [@gupta_2013; @huang_2009]. Despite their relative success, the treatments are more short term and do not actively address the problem of immune dysfunction within the skin. Recent therapies such as Ustekinumab have be introduced that specifically target pathways in the immune response to avoid this overactive immune behavior, and consequently stop the lesions all together [@McKeage2014].

The pathophysiology of psoriasis is heavily mediated by the immune system, specifically T cell. Keratinocytes are triggered by a factors such as stress, which induces them to release cytokines [@lande2007]. These cytokines then activate T cells, namely Th1 and Th17, which further enhance this immune response by releasing more cytokines that introduces other immune cells. This activation leads to inflammation within the lesion microenvironment, which further triggers keratinocytes to start actively proliferating, a response usually induced to shed possible pathogens. Since this initiation of the immune response is heavily initiated by T cells, studying these cells are a major area of interest. Specifically what may be of interest is how these cells are affected in terms of their typical cellular functions. Many cellular functions in the immune system have been researched as being key to immune function. Changes in these critical cellular processes may be key to the dysfunction seen in many autoimmune disease, including psoriasis. Cell adhesion is a key process that enables the physical movement of immune cells into the blood vessels and to sites of inflammation via cell receptors such as selectins [@afanasiev2013]. Cytokines and their chemokine subsets are a major signaling proteins in the immune system that can trigger inflammation and homing of immune cells to specific sites [@baliwag2015; @lee2012]. Inhibitory mechanisms such as CTLA4 and PD1 are major receptors involved in inhibition of immune cells in disease like cancer, dampening the immune system [@hernandez2011; @krummel_1995]. Lastly, cell proliferation via clonal expansion of immune such as T and B cells are what largely drive the inflammation as more inflammatory cells are produced [@denizot1986; @prakken2000]. Taken together, these functions are critical in the immune response, and any dysfunction in these would likely have some degree of effect in normal immune function in disease such as psoriasis.

Given that there are so many key processes used by the immune system, we decide that we wanted to look closer at these in psoriasis. Specifically we wanted to see how they were affected in psoriasis, and if there were any major changes between subtypes. In order to investigate this, we looked at major differences in these processes on the genetic level. We did so by using the RNA sequencing data originally generated by [@ahn2018] from active psoriasis lesions and healthy control. They specifically looked at the psoriasis lesions from patients with conventional psoriasis, palmoplantar psoriasis, and scalp psoriasis to look for any molecular changes. All of these subtypes differ mainly in the location to which the lesions are expressed, indicating some likely different immune dysfunction phenotypes. Moving forward with this idea, we looked at how cellular processes were affected in psoriasis in general. We then honed in specifically on the major process affected and sought to understand how these differed specifically across the different psoriasis types. We hypothesized that cell adhesion mechanisms will be reduced, likely in regulatory populations, and migration, inhibition, inflammation, and proliferation mechanisms will be increased, likely in the effector and regulatory populations, in psoriasis so as to favor the autoimmune response. We further built on this and hypothesized that the degree to which these are affected will different between psoriasis type so as favor the phenotype that is key in that specific disease. All together, we aim to gain better understanding in the processes affected in psoriasis and how these may be slightly altered between different subtypes. We hope that these finding may shed some more light onto the exact mechanisms affected in each subtype so that more targeted therapies can be created to specifically treat the exact dysfunctionalities in each subtype.


# Methods

## Study design

The original aim [@ahn2018] was to identify the heterogeneity on molecular level with in psoriasis patients. They did so by using RNA-Seq in order to identify the differences in gene expression as well as flow cytometry to quantify T cell make-up. Their study essentially consisted of isolating skin samples from psoriasis patients and healthy control skin to extract the RNA and analyze the cells via flow cytometry. The design consisted of 3 experimental groups: scalp psoriasis, palmoplantar psoriasis, and conventional plaque psoriasis. A control group was included as well using a healthy control group. Two 4 mm samples were taken from the active psoriatic lesions of the experimental group skin and from the healthy control skin. For each skin sample, 1 biopsy was used for RNA-Seq analysis and the other was used for flow cytometry analysis. The individuals from which the psoriatic skin was obtained were at least 18 years old and were confirmed to have psoriasis for a minimum of 6 months. Individuals with psoriasis were not on systemic treatments and abstained from topical treatments for at least 2 weeks so as to not interfere with the data. The study design allowed for the comparison of all psoriasis types between each other as well as with healthy controls 

## Sample origin and sequencing

For the experimental groups, a total of 19 patients with psoriasis were recruited from the San Francisco Bay area. Of these patients, 8 had conventional plaque psoriasis, 8 had scalp psoriasis, and 3 had palmoplantar psoriasis. Two 4 mm punch biopsies were taken from active lesions on the trunk and extremities of the conventional plaque psoriasis, on the scalp of scalp psoriasis patients, and on the hand or foot of the palmoplantar psoriasis patients. For the control group, skin was isolated from the trunk or upper legs of 9 patients. The healthy control skin was obtained from the surgical discards as a results of cosmetic surgery performed on these patients. The biopsies for RNA-Seq were placed in RNALater to preserve the RNA. The biopsies for flow cytometry were placed on gauze dampened with phosphate buffered saline to keep the skin hydrated. All biopsies were then stored at 4 Â°C until processing of the samples. 

The RNA was extracted from the biopsies started in RNALater using first the Qiagen Allprep DNA/RNA mini kit. Once extracted, quality control and identification of mRNA with little degradation was performed on the Agilent 2100 bioanalyzer in order to filter out any suboptimal mRNA that may interfere with analysis. The quality mRNA that was not discarded was then prepared into a library either the Ovation RNA-Seq System V2 protocol (Nugen) or the Ribozero (Illumina) kits. The prepared libraries were then sequenced on the Illumina Hiseq platform [@metzker2010sequencing]. The reads obtained were paired-end and maintained a 100 bp length. 


## Computational

The data from [@ahn2018] used for this analysis were download from the Gene Expression Omnibus using the accession number GSE117405 using the SRA toolkit. The FASTQ files were then used to perform quality control of the reads within the files based on the quality scores assigned to the reads. This allowed for the visualization of suboptimal quality regions of reads. The trimmomatic program was used to then identify these lower quality regions of reads and trim them if their quality score begins to fall below the threshold of 15 [@bolger2014]. Once the reads were cleaned to avoid errors from read sequences, the data was then ready for further processing. The human reference genome, transcripts, and annotation were downloaded from the NCBI RefSeq database in order to align the trimmed reads to the genome in order to identify the genes expressed. The biomartr program was used to download these data as it allows for a controlled and reproducible manner of retrieving such information [@drost2017]. Once this data was obtained, the reads were ready to be aligned. The sailfish package was used in order to index the reads and align them to the provided reference genome [@patro2014]. Counts of these aligned reads were then generated using the same package. These aligned transcripts were then identified via their GenBank IDs, which were used to obtain the gene names of these transcripts. Counts were ultimately consolidated into a singled gene count table containing normalized and non-normalized gene counts which could be used for analysis. 

The prepared data table was then analyzed in RStudio to produce the graphs here. Data was manipulated to look at expression of genes of interest via normalized counts across the conditions to see changes in psoriasis and the types of psoriasis. The non-normalized counts were also used to perform differential analysis via DESeq2 for figure 2 in order to cluster samples based on the top 1000 gene differentially expressed genes [@Love2014].


# Results

## Subsections are ok in the results section too

```{r your-code-here, include = FALSE}
# Add code chunks as needed for your analyses
# For most analyses, I would recommend splitting the
# intensive computational part into a seperate R script file
# and then just load your libraries and the data object here
# using the `load()` function. Ask me for clarification if this is
# unclear.
```

```{r load-packages-and-data, include = FALSE}
# load packages as needed
library("plyr")
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("magrittr")
library("tibble")
library("DESeq2")
library("RColorBrewer")
library("ggfortify")
library("cluster")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# load data produced from analysis scripts using
# something like load("output/processed_data.Rdata")
load("output/final_compiled_counts/joined_count_data.RData")

# test that it loaded correctly before proceeding
stopifnot(exists("joined_table"))

# load data produced from differential expression analysis script
load("output/deseq2_data/rld_deseq2_data.RData")

# test that it loaded correctly before proceeding
stopifnot(exists("rld"))

```

We first wanted to identify the major cell mechanisms affected in psoriasis in general, specifically cell adhesion, cytokines, chemokines, inhibition, and proliferation. We looked at the gene sets involved in these mechanisms to see which mechansims were upregulated or downregulated. We calculate fold changes in expression to see how upregulated or downregulated certain processes were in psoriasis in general. 

```{r figure-1-Enrichment-Plot, fig.width = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  mutate(psoriasis_or_healthy = recode(disease_state,
                                       "psoriasis" = "psoriasis",
                                       "palmoplantar psoriasis" = "psoriasis",
                                       "scalp psoriasis" = "psoriasis",
                                       "healthy control" = "healthy")) %>%
  mutate(gene_function = recode(genename,
                                "MKI67" = "Proliferation",
                                "CDK1" = "Proliferation",
                                "CDK2" = "Proliferation",
                                "CDK4" = "Proliferation",
                                "CDK6" = "Proliferation",
                                "CDK7" = "Proliferation",
                                "CDK8" = "Proliferation",
                                "CDK9" = "Proliferation",
                                "CDK3" = "Proliferation",
                                "PCNA" = "Proliferation",
                                "TOP2A" = "Proliferation",
                                "MCM2" = "Proliferation",
                                "MCM3" = "Proliferation",
                                "MCM4" = "Proliferation",
                                "MCM5" = "Proliferation",
                                "MCM6" = "Proliferation",
                                "MCM7" = "Proliferation",
                                "MCM8" = "Proliferation",
                                "MCM9" = "Proliferation",
                                "ITGA7" = "Cell Adhesion Molecules",
                                "ITGB1" = "Cell Adhesion Molecules",
                                "ITGA6" = "Cell Adhesion Molecules",
                                "ITGB2" = "Cell Adhesion Molecules",
                                "ITGAV" = "Cell Adhesion Molecules",
                                "ITGB3" = "Cell Adhesion Molecules",
                                "ITGA2" = "Cell Adhesion Molecules",
                                "ITGA4" = "Cell Adhesion Molecules",
                                "ITGA3" = "Cell Adhesion Molecules",
                                "ITGA5" = "Cell Adhesion Molecules",
                                "ITGA2B" = "Cell Adhesion Molecules",
                                "ITGB4" = "Cell Adhesion Molecules",
                                "ITGAM" = "Cell Adhesion Molecules",
                                "ITGAL" = "Cell Adhesion Molecules",
                                "ITGA1" = "Cell Adhesion Molecules",
                                "ITGAE" = "Cell Adhesion Molecules",
                                "ITGA9" = "Cell Adhesion Molecules",
                                "ITGAX" = "Cell Adhesion Molecules",
                                "ITGB7" = "Cell Adhesion Molecules",
                                "ITGA11" = "Cell Adhesion Molecules",
                                "ITGB6" = "Cell Adhesion Molecules",
                                "ITGA8" = "Cell Adhesion Molecules",
                                "ITGB5" = "Cell Adhesion Molecules",
                                "ITGAD" = "Cell Adhesion Molecules",
                                "ITGA10" = "Cell Adhesion Molecules",
                                "VCAM1" = "Cell Adhesion Molecules",
                                "ICAM1" = "Cell Adhesion Molecules",
                                "ICAM2" = "Cell Adhesion Molecules",
                                "ICAM3" = "Cell Adhesion Molecules",
                                "ICAM4" = "Cell Adhesion Molecules",
                                "ICAM5" = "Cell Adhesion Molecules",
                                "SELP" = "Cell Adhesion Molecules",
                                "SELL" = "Cell Adhesion Molecules",
                                "ICAM5" = "Cell Adhesion Molecules",
                                "CXCR1" = "Chemokine Receptors",
                                "CXCR2" = "Chemokine Receptors",
                                "CXCR3" = "Chemokine Receptors",
                                "CXCR4" = "Chemokine Receptors",
                                "CXCR5" = "Chemokine Receptors",
                                "CXCR6" = "Chemokine Receptors",
                                "CXCR7" = "Chemokine Receptors",
                                "CXCR4" = "Chemokine Receptors",
                                "CCR1" = "Chemokine Receptors",
                                "CCR2" = "Chemokine Receptors",
                                "CCR3" = "Chemokine Receptors",
                                "CCR4" = "Chemokine Receptors",
                                "CCR5" = "Chemokine Receptors",
                                "CCR6" = "Chemokine Receptors",
                                "CCR7" = "Chemokine Receptors",
                                "CCR8" = "Chemokine Receptors",
                                "CCR9" = "Chemokine Receptors",
                                "CCR10" = "Chemokine Receptors",
                                "CCR11" = "Chemokine Receptors",
                                "CXCL1" = "Chemokines",
                                "CXCL2" = "Chemokines",
                                "CXCL3" = "Chemokines",
                                "CXCL4" = "Chemokines",
                                "CXCL5" = "Chemokines",
                                "CXCL6" = "Chemokines",
                                "CXCL7" = "Chemokines",
                                "CXCL8" = "Chemokines",
                                "CXCL9" = "Chemokines",
                                "CXCL10" = "Chemokines",
                                "CXCL11" = "Chemokines",
                                "CXCL12" = "Chemokines",
                                "CXCL13" = "Chemokines",
                                "CXCL14" = "Chemokines",
                                "CXCL15" = "Chemokines",
                                "CXCL16" = "Chemokines",
                                "CCL1" = "Chemokines",
                                "CCL2" = "Chemokines",
                                "CCL3" = "Chemokines",
                                "CCL4" = "Chemokines",
                                "CCL5" = "Chemokines",
                                "CCL6" = "Chemokines",
                                "CCL7" = "Chemokines",
                                "CCL8" = "Chemokines",
                                "CCL9" = "Chemokines",
                                "CCL10" = "Chemokines",
                                "CCL11" = "Chemokines",
                                "CCL12" = "Chemokines",
                                "CCL13" = "Chemokines",
                                "CCL14" = "Chemokines",
                                "CCL15" = "Chemokines",
                                "CCL16" = "Chemokines",
                                "CCL17" = "Chemokines",
                                "CCL18" = "Chemokines",
                                "CCL19" = "Chemokines",
                                "CCL20" = "Chemokines",
                                "CCL21" = "Chemokines",
                                "CCL22" = "Chemokines",
                                "CCL23" = "Chemokines",
                                "CCL24" = "Chemokines",
                                "CCL26" = "Chemokines",
                                "CCL27" = "Chemokines",
                                "CCL28" = "Chemokines",
                                "CCL1" = "Chemokines",
                                "TNF" = "Proinflammatory Cytokines",
                                "TNFRSF1A" =
                                  "Proinflammatory Cytokines Receptors",
                                "TNFRSF1B" =
                                  "Proinflammatory Cytokines Receptors",
                                "TRAF1" = "Proinflammatory Cytokines Receptors",
                                "TRAF2" = "Proinflammatory Cytokines Receptors",
                                "TRAF3" = "Proinflammatory Cytokines Receptors",
                                "TRAF4" = "Proinflammatory Cytokines Receptors",
                                "TRAF5" = "Proinflammatory Cytokines Receptors",
                                "TRAF6" = "Proinflammatory Cytokines Receptors",
                                "IL1B" = "Proinflammatory Cytokines",
                                "IL1A" = "Proinflammatory Cytokines",
                                "IL1R1" = "Proinflammatory Cytokines Receptors",
                                "IL2" = "Proinflammatory Cytokines",
                                "IL2RA" = "Proinflammatory Cytokines Receptors",
                                "IL2RB" = "Proinflammatory Cytokines Receptors",
                                "IL2RG" = "Proinflammatory Cytokines Receptors",
                                "IL6" = "Proinflammatory Cytokines",
                                "IL6R" = "Proinflammatory Cytokines Receptors",
                                "IL12A" = "Proinflammatory Cytokines",
                                "IL12B" =
                                  "Proinflammatory Cytokines",
                                "IL12RB1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL12RB2" =
                                  "Proinflammatory Cytokines Receptors",
                                "IFNG" =
                                  "Proinflammatory Cytokines",
                                "IFNA1" =
                                  "Proinflammatory Cytokines",
                                "IFNGR1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IFNB1" =
                                  "Proinflammatory Cytokines",
                                "IFNAR1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL18" =
                                  "Proinflammatory Cytokines",
                                "IL18R1" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL18RAP" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL18BP" = "Proinflammatory Cytokines",
                                "IL7" = "Proinflammatory Cytokines",
                                "IL9" = "Proinflammatory Cytokines",
                                "IL10" = "Proinflammatory Cytokines",
                                "IL7R" = "Proinflammatory Cytokines",
                                "IL9R" = "Proinflammatory Cytokines",
                                "IL10RB" = "Proinflammatory Cytokines",
                                "IL10RA" = "Proinflammatory Cytokines",
                                "IL17A" =
                                  "Proinflammatory Cytokines",
                                "IL17B" =
                                  "Proinflammatory Cytokines",
                                "IL17C" = "Proinflammatory Cytokines",
                                "IL17D" = "Proinflammatory Cytokines",
                                "IL17F" =
                                  "Proinflammatory Cytokines",
                                "IL17RA" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL17RB" =
                                  "Proinflammatory Cytokines Receptors",
                                "IL17RC" =
                                  "Proinflammatory Cytokines Receptors",
                                "CD80" = "Inhibitory Cytokines and Ligands",
                                "CD86" = "Inhibitory Cytokines and Ligands",
                                "CD274" = "Inhibitory Cytokines and Ligands",
                                "BTLA" = "Inhibitory Cytokines and Ligands",
                                "CD160" = "Inhibitory Cytokines and Ligands",
                                "PVAR" = "Inhibitory Cytokines and Ligands",
                                "CD48" = "Inhibitory Cytokines and Ligands",
                                "LGALS9" = "Inhibitory Cytokines and Ligands",
                                "CTLA4" = "Inhibitory Receptors",
                                "PDCD1" = "Inhibitory Receptors",
                                "HAVCR2" = "Inhibitory Receptors",
                                "LAG3" = "Inhibitory Receptors",
                                "TIGIT" = "Inhibitory Receptors",
                                "CD244" = "Inhibitory Receptors")) %>%
  filter(gene_function %in% c("Proliferation",
                              "Cell Adhesion Molecules",
                              "Chemokine Receptors",
                              "Proinflammatory Cytokines Receptors",
                              "Chemokines",
                              "Proinflammatory Cytokines",
                              "Inhibitory Cytokines and Ligands",
                              "Inhibitory Receptors")) %>%
  group_by(gene_function, psoriasis_or_healthy) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(psoriasis_or_healthy, avg_count) %>%
  transform(fold_expression = psoriasis / healthy) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = gene_function,
             y = inverse_or_return,
             fill = gene_function)) +
  geom_col() +
  coord_flip() +
  scale_fill_brewer(palette = "Spectral", name = "Gene Set") +
  geom_hline(yintercept = 0,
                color = "grey35", size = 1) +
  geom_hline(yintercept = -1,
                color = "grey35", size = .5, linetype = "dashed") +
  geom_hline(yintercept = 1,
                color = "grey35", size = .5, linetype = "dashed") +
  ggtitle("Fold Change in Expression of Gene Sets
\nbetween Psoriasis and Healthy skin") +
  xlab("Gene Set") +
  ylab("Fold Change")
```

**Figure 1**: Bar plot of major gene set involved in key cellular processes that are upregulated or downregulated in psoriasis.

We first found that the expression levels of most of these cellular processes were largely affected, except for two. Specifically, proliferation, inhibition, and chemokines were all largely upregulated (figure 1). Meanwhile, cell adhesion was one process that was largely downregulated (figure 1). Proinflammatory cytokines trended towards downregulation and proinflammatory cytokine receptors trended towards upregulation, but not enough to keep moving forward with these (figure 1). Using this information, we decided to look more closely at those processes that were largely affected across the different psoriasis types.

Before moving forward and looking more in-depth at these cellular processes, we decided to look more closely into the differen subtypes in general. We wnated to gain more insight into these subtypes and gain some more information on exactly how they relate. More specifically, we wanted to see how exactly similar they were and if they showed some distinct phenotypes.

```{r figure-2-PCA-Plot, fig.align = "center", echo = FALSE}
# Create PCA plot comparing disease type groups based on the top 1000 expressed genes
pca <- plotPCA(rld,
               intgroup = c("disease_state"),
               ntop = 1000,
               returnData = TRUE)
percent_var <- round(100 * attr(pca, "percentVar"))
find_hull <- function(pca) pca[chull(pca$PC1, pca$PC2), ]
hulls <- ddply(pca, "disease_state", find_hull)
ggplot(pca, aes(PC1, PC2, color = disease_state)) +
        geom_point(size = 3) +
        xlab(paste0("PC1: ", percent_var[1], "% variance")) +
        ylab(paste0("PC2: ", percent_var[2], "% variance")) +
  ggtitle("Comparison of the Psoriasis Disease \nTypes and Healthy Control") +
  geom_polygon(aes(fill = disease_state), data = hulls, alpha = 0.3) +
  guides(fill = FALSE) +
  scale_color_discrete(name = "Condition")
```

**Figure 2**: PCA plot of the 4 sample groups to infer how the phenotypes of psoriasis differ based on the top 100 differentially expressed genes. 

In order to gain some idea of relatedness between the psoriasis subtypes, we used a principal component analysis approach to calculate the differential expression of the top 1000 variably expressed genes. Using this allowed us to infer some idea of relatedness based on the difference in expression of certain genes. We saw that there was some degree of agregation across the psoriasis subtypes (Figure 2). The healthy controls congregated cleanly together (Figure 2). The subtypes congregated to some extent but there was some overlap between conventional psoriasis and scalp psoriasis (Figure 2). Moreover, the points within each group varied largely on the PC1 axis (Figure 2).

Now that we had some idea of relatedness, we then moved forward as stated before. We bgan by looking at the upregulation or down regulation via fold change acorss the different subtypes to see how these chnaged. We wanted to see if there were many changes from one subtypes to another.

```{r figure-3-heatmap-inhibitory-cytokines-and-ligands, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

hm_palette <- colorRampPalette(rev(brewer.pal(9, "Spectral")), space = "Lab")

joined_table %>%
  filter(genename %in% c("CD80",
                         "CD86",
                         "CD274",
                         "BTLA",
                         "CD160",
                         "PVAR",
                         "CD48",
                         "LGALS9")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp =
              `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type,
         fold_expression,
         fold_expression_pp:fold_expression_p) %>%
  select(genename,
         palmoplantar.psoriasis,
         psoriasis, scalp.psoriasis,
         fold_change_type,
         fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Inhibitory Cytokine
\nand Ligand Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change",
                       colours = hm_palette(100),
                       limits = c(-9, 10)) +
  scale_x_discrete(labels = c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```

**Figure 3**: Heatmap looking at the fold change expression of inhibitory ligands and cytokines amongst the psoriasis subtypes.

We began by looking at the fold change expression of inhibitory ligands and cytokines to see how their expression changed. We found that there was an overall upregulation as in figure 1 (Figure 3). There was some degree of variability in the expression of this upregulation, however, for many the genes (Figure 3). Additionally, there were some genes that were downregulated overall such as CD160 and even one gene that was slightly downregulated in only palmoplantar psoriasis, LGALS9 (Figure 3).

```{r figure-4-heatmap-inhibitory-receptors, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("CTLA4",
                         "PDCD1",
                         "HAVCR2",
                         "LAG3",
                         "TIGIT",
                         "CD244")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp =
              `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type,
         fold_expression,
         fold_expression_pp:fold_expression_p) %>%
  select(genename,
         palmoplantar.psoriasis,
         psoriasis,
         scalp.psoriasis,
         fold_change_type,
         fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Inhibitory
\nReceptor Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change",
                       colours = hm_palette(100),
                       limits = c(-5, 5)) +
  scale_x_discrete(labels = c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```

**Figure 4**: Heatmap looking at the fold change expression of inhibitory receptors amongst the psoriasis subtypes.

We then moved to look at the fold change expression of inhibitory receptors across the subtypes. We found an overall upregulation as well, similar to figure 1 (Figure 4). There was much less variability in the degree of upregulation of these genes across the subtypes (Figure 4). Despite this, HAVCR2 was one gene that was slightly downregulated in only palmoplantar psoriasis and scalp psoriasis (Figure 4).

```{r figure-5-heatmap-chemokine-receptors, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("CXCR1",
                         "CXCR2",
                         "CXCR3",
                         "CXCR4",
                         "CXCR5",
                         "CXCR6",
                         "CXCR7",
                         "CXCR4",
                         "CCR1",
                         "CCR2",
                         "CCR3",
                         "CCR4",
                         "CCR5",
                         "CCR6",
                         "CCR7",
                         "CCR8",
                         "CCR9",
                         "CCR10",
                         "CCR11")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp =
              `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type,
         fold_expression,
         fold_expression_pp:fold_expression_p) %>%
  select(genename,
         palmoplantar.psoriasis,
         psoriasis,
         scalp.psoriasis,
         fold_change_type,
         fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Chemokine
\nReceptor Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change",
                       colours = hm_palette(100),
                       limit = c(-6.5, 6.5)) +
  scale_x_discrete(labels = c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```

**Figure 5**: Heatmap looking at the fold change expression of chemokine receptors amongst the psoriasis subtypes.

After inhibition, we started to analyze expression of chemokines, specfically their receptors. There was mainly upregulation of these genes for all subtypes, but there was also some downregulation of certain genes as well such as CCR10, CCR9, and CXCR5 that was seen for all three (figure 5). Again, the degree to which these were expressed varied across subtype (figure 5). Additionally, CCR3, CCR6, and CCR8 were some receptors that were down regulated in only specific subtypes (figure 5). 

```{r figure-6-heatmap-chemokines, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("CXCL1",
                                "CXCL2",
                                "CXCL3",
                                "CXCL4",
                                "CXCL5",
                                "CXCL6",
                                "CXCL7",
                                "CXCL8",
                                "CXCL9",
                                "CXCL10",
                                "CXCL11",
                                "CXCL12",
                                "CXCL13",
                                "CXCL14",
                                "CXCL15",
                                "CXCL16",
                                "CCL1",
                                "CCL2",
                                "CCL3",
                                "CCL4",
                                "CCL5",
                                "CCL6",
                                "CCL7",
                                "CCL8",
                                "CCL9",
                                "CCL10",
                                "CCL11",
                                "CCL12",
                                "CCL13",
                                "CCL14",
                                "CCL15",
                                "CCL16",
                                "CCL17",
                                "CCL18",
                                "CCL19",
                                "CCL20",
                                "CCL21",
                                "CCL22",
                                "CCL23",
                                "CCL24",
                                "CCL26",
                                "CCL27",
                                "CCL28",
                                "CCL1")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp =
              `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type,
         fold_expression,
         fold_expression_pp:fold_expression_p) %>%
  select(genename,
         palmoplantar.psoriasis,
         psoriasis,
         scalp.psoriasis,
         fold_change_type,
         fold_expression) %>%
  gather(disease_state, avg_count, palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Chemokine \nGenes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change",
                       colours = hm_palette(100),
                       limits = c(-40, 60)) +
  scale_x_discrete(labels = c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```

**Figure 6**: Heatmap looking at the fold change expression of chemokines amongst the psoriasis subtypes.

We then analyzed expression of chemokines for these chemokine receptors. There was very minimal downregulation of all chemokines looked at for all subtypes, with some variability (figure 6). However, there were some major upregulated chemokines for specific subtypes. CXCL10 and CXCL9 were upregulated mainly in palmoplantar psoriasis while CXCL13 was mainly upregulated in conventianl psoriasis (figure 6). There was also some upregulation of certain chemokines for all subtypes, but to a much less extent than these previous ones mentioned (figure 6).

```{r figure-7-heatmap-adhesion, fig.height = 10, fig.align = "center", echo = FALSE}
inverse_func <- function(x) {
  if (x >= 1) {
    return(x)
  } else {
    return(-1 / x)
  }
}

joined_table %>%
  filter(genename %in% c("ITGA7",
                                "ITGB1",
                                "ITGA6",
                                "ITGB2",
                                "ITGAV",
                                "ITGB3",
                                "ITGA2",
                                "ITGA4",
                                "ITGA3",
                                "ITGA5",
                                "ITGA2B",
                                "ITGB4",
                                "ITGAM",
                                "ITGAL",
                                "ITGA1",
                                "ITGAE",
                                "ITGA9",
                                "ITGAX",
                                "ITGB7",
                                "ITGA11",
                                "ITGB6",
                                "ITGA8",
                                "ITGB5",
                                "ITGAD",
                                "ITGA10",
                                "VCAM1",
                                "ICAM1",
                                "ICAM2",
                                "ICAM3",
                                "ICAM4",
                                "ICAM5",
                                "SELP",
                                "SELL",
                                "ICAM5")) %>%
  group_by(genename, disease_state) %>%
  summarise(avg_count = mean(counts_lengthscaledtpm)) %>%
  spread(disease_state, avg_count) %>%
  transform(fold_expression_pp =
              `palmoplantar psoriasis` / `healthy control`) %>%
  transform(fold_expression_s = psoriasis / healthy.control) %>%
  transform(fold_expression_p = scalp.psoriasis / healthy.control) %>%
  gather(fold_change_type,
         fold_expression,
         fold_expression_pp:fold_expression_p) %>%
  select(genename,
         palmoplantar.psoriasis,
         psoriasis,
         scalp.psoriasis,
         fold_change_type,
         fold_expression) %>%
  gather(disease_state,
         avg_count,
         palmoplantar.psoriasis:scalp.psoriasis) %>%
  rowwise() %>%
  mutate(inverse_or_return = inverse_func(fold_expression)) %>%
  ggplot(aes(x = fold_change_type,
             y = genename)) +
  geom_tile(aes(fill = inverse_or_return)) +
  xlab("Psoriasis") +
  ylab("Gene") +
  ggtitle("Fold Change Expression of Cell
\nAdhesion Genes in Psoriasis Types") +
  scale_fill_gradientn(name = "Fold Expression Change",
                       colours = hm_palette(100),
                       limits = c(-6, 6)) +
  scale_x_discrete(labels = c("fold_expression_p" = "Conventional \nPsoriasis",
                            "fold_expression_pp" = "Palmoplantar \nPsoriasis",
                            "fold_expression_s" = "Scalp \nPsoriasis"))
```
 
**Figure 7**: Heatmap looking at the fold change expression of adhesion molecules amongst the psoriasis subtypes.

We finally looked at the expression of adhesion molecules and found that there was a mix of upregulation and downregulation, but downregualtion was more prominent (figure 7). These expression levels were seen similarly across all the subtypes with some variability (figure 7). There were some genes such as ICAM5 and ITGA10 that were downregulated in only palmoplantar psoriasis and scalp psoriasis that was not seen in conventional psoriasis (figure 7). 

# Discussion


# Sources Cited
